katsdpimager/loader_katdal.py

```diff
@@ -206,12 +206,15 @@ def __init__(self, filename, options):
                 band_mask = fetcher.get(band_mask_key, katsdpmodels.band_mask.BandMask,
                                         telstate=telstate_acv)
                 channel_freqs = self._file.channel_freqs * units.Hz
-                max_length = rfi_mask.max_baseline_length(channel_freqs)
                 spw = katsdpmodels.band_mask.SpectralWindow(
                     telstate_acv['bandwidth'] * units.Hz,
                     telstate_acv['center_freq'] * units.Hz
                 )
-                self._channel_mask = (max_length > 0) | band_mask.is_masked(spw, channel_freqs)
+                channel_width = spw.bandwidth / telstate_acv['n_chans']
+                max_length = rfi_mask.max_baseline_length(channel_freqs, channel_width)
+                self._channel_mask = (
+                    (max_length > 0) | band_mask.is_masked(spw, channel_freqs, channel_width)
```

Comment:
Does `max_length == 0` mean that the autocorrelations are flagged in that channel? That suggests `max_length >= 0` to me.

Reply:
There is a separate flag in rfi_mask which determines whether autocorrelations should be flagged. But it doesn't matter whether autocorrelations are flagged because they're not imaged.

Reply:
And hopefully they don't affect the weights either :-)

Reply:
So what does `max_baseline_length == 0` mean? It is not spelled out in the docs (the one I have to port to SKA soon 😂 ).

If you already have a `mask_auto_correlations` property, why not use 0 to indicate "no masking"?

Reply:
I think it would mean that a model file has requested baselines of up to length zero (i.e., only autos) should be flagged for some frequency range.

Reply:
Would you ever flag the crosses and not the autos?

Reply:
I think that's the current behaviour. I can't remember why; @KimMcAlpine might be able to tell you since that was what she originally implemented in katsdpcal.

Reply:
Unless we turn the description to "maximum non-zero baseline length ... ". That would clear things up and allow 0 to be a special case.

This won't change any user code of this API, since you already check `max_length > 0` in all cases. It would just prevent someone from using it for autos.

Reply:
Are there any changes you want **in this PR**?